util/paused/paused_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package paused implements paused helper functions.
+package paused
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/internal/test/builder"
+)
+
+func TestEnsurePausedCondition(t *testing.T) {
+	g := NewWithT(t)
+
+	scheme := runtime.NewScheme()
+	g.Expect(builder.AddTransitionV1Beta2ToScheme(scheme)).To(Succeed())
+	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
+
+	// Cluster Case 1: unpaused
+	normalCluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "some-cluster",
+			Namespace: "default",
+		},
+	}
+
+	// Cluster Case 2: paused
+	pausedCluster := normalCluster.DeepCopy()
+	pausedCluster.Spec.Paused = true
+
+	// Object case 1: unpaused
+	obj := &builder.Phase1Obj{ObjectMeta: metav1.ObjectMeta{
+		Name:      "some-object",
+		Namespace: "default",
+	}}
+
+	// Object case 2: paused
+	pausedObj := obj.DeepCopy()
+	pausedObj.SetAnnotations(map[string]string{clusterv1.PausedAnnotation: "true"})
```

Comment:
```suggestion
	pausedObj.SetAnnotations(map[string]string{clusterv1.PausedAnnotation: ""})
```
nit: the check doesn't depend on any specific value, let's align the unit test accordingly